### ISSUE TITLE
[BUG] TypeError: Incompatible fields types while JSON.stringify

### ISSUE THREAD

**Issue by viceice** [nilproject/NiL.JS#224]
When trying to serialize as `class` with 2 `indexer` it will throw a `JSException`.

See https://github.com/viceice/niljs-testing.git for repro.

https://github.com/viceice/niljs-testing/blob/db376bed4f9336ecf1c577252511055f7695b58a/test/Tests.cs#L77-L101

```cs
        class IndexerDummy
        {

            public object this[string key] { get => key; }
            public object this[Guid key] { get => key; }

            public string Title => "title";

            public bool Update() => true;
        }
```

```
Message: 
    NiL.JS.Core.JSException : TypeError: Incompatible fields types.
  Stack Trace: 
    MethodProxy.invokeMethod(JSValue targetValue, Expression[] argumentsSource, Arguments argumentsObject, Context initiator)
    MethodProxy.InternalInvoke(JSValue targetValue, Expression[] argumentsSource, Context initiator, Boolean withSpread, Boolean withNew)
    Call.Evaluate(Context context)
    CodeBlock.evaluateLines(Context context, Int32 i, Boolean clearSuspendData)
    CodeBlock.Evaluate(Context context)
    Context.Eval(String code, JSValue thisBind, Boolean suppressScopeCreation)
    Context.Eval(String code, Boolean suppressScopeCreation)
    Tests.Test_Issue_224_json() line 99
```

**Comment by viceice** [comment]
Serializing the following causes `AmbiguousMatchException` for duplicate `Item` property on `Base2`
```cs

        abstract class Base
        {
            public abstract object this[string key] { get; set;  }

        }
        abstract class Base2: Base
        {
            public abstract Dummy this[Guid key] { get; }

        }
        private class Dummy: Base2
        {
            public int Id => 5;
            public string Title => "title";

            public override object this[string key] { get => key; set { } }
            public override Dummy this[Guid key] => null;

            public object ToJson() => string.Empty;
            public object ToJson(string key) => key;
        }
```

Exception is thrown here:
https://github.com/nilproject/NiL.JS/blob/16429fca7f44e129235098576ae89aebc4351501/NiL.JS/Core/Interop/Proxy.cs#L151

What is the cause for going to base types? Can't we simply use the property of current type?

```
 Message: 
    Test method IntegrationTests.BaseLibrary.JsonTests.CustomSerializer threw exception: 
    System.Reflection.AmbiguousMatchException: Ambiguous match found.
  Stack Trace: 
    RuntimeType.GetPropertyImpl(String name, BindingFlags bindingAttr, Binder binder, Type returnType, Type[] types, ParameterModifier[] modifiers)
    Type.GetProperty(String name, BindingFlags bindingAttr)
    RuntimeReflectionExtensions.GetRuntimeProperty(Type type, String name)
    Proxy.fillMembers() line 151
    Proxy.GetProperty(JSValue key, Boolean forWrite, PropertyScope memberScope) line 333
    JSObject.GetProperty(JSValue key, Boolean forWrite, PropertyScope propertyScope) line 145
    ObjectWrapper.GetProperty(JSValue name, Boolean forWrite, PropertyScope memberScope) line 58
    JSValue.GetProperty(String name) line 472
    JSValue.get_Item(String name) line 212
    JSON.stringifyImpl(String key, JSValue obj, Function replacer, HashSet`1 keys, String space, HashSet`1 processed, Arguments args) line 477
    JSON.stringify(JSValue obj, Function replacer, HashSet`1 keys, String space) line 368
    JSON.stringify(Arguments args) line 359
    JSON.stringify(JSValue value) line 307
```

**Comment by viceice** [comment]
Loop was introduced here:
https://github.com/nilproject/NiL.JS/blob/4651751429c497b579975685ae157c3234185e55/NiL.JS/Core/Interop/Proxy.cs#L148-L157

added as fix for #92
